libs/tree/ui/src/lib/selection-tree/selection-tree.component.ts

```diff
@@ -103,7 +103,7 @@ export class SelectionTreeComponent<T>
     this.nodeClick.emit(nodeId);
     this._transformer.updateOne(nodeId);
     this.states = this._transformer.states;
-    this.model = this._transformer.array;
+    this.model = this._transformer.set;
```

Comment:
This is a critical line that needs to change, but changing it with no further adjustments breaks consuming apps that were built to expect an array model to come out of the component.

The intention is to make the component so that it can accept an array OR set model and future model updates will be issued with that same type for uniformity: same data type going in and coming out, and the component is intelligent enough to understand which one to output.

Likely the best solution will be to add additional functionality to the `Transformer` class so that it registers the last-used model type (whether on initialization or updates) and can return the updated model of the same type via a new getter (probably called `.model`). This way, the component doesn't need to manage which type to ask for from the `Transformer`: it can simply call `.update()` followed by `.model()` and be confident it's getting back a model it can use